### PR TITLE
Build fixes for GCC2 and GCC7 WIP.

### PR DESCRIPTION
**Pull request by waddlesplash**

### CODE REVIEW

**Comment by waddlesplash** [comment]
Only one functional change (that I know of): added braces around an `if` that needed them, spotted by GCC7's -Wmisleading-indentation.

**Comment by waddlesplash** [comment]
OK, I had to hack up a `libroot` to work around https://github.com/diversys/liblayout/issues/2, and then add another hack to agg to work around https://dev.haiku-os.org/ticket/14531, but GCC7 WonderBrush seems to work just fine.

**Comment by stippi** [comment]
Wow, thanks! I'm diving right into the diff...

File: WonderBrush/src/gui/panels/about/EdgesFilter.cpp [hunk 1]
```diff
@@ -54,8 +54,8 @@ EdgesFilter::Run(const BBitmap* input, const BBitmap* output) const
 						top = (src[pos] + src[pos + 1] * 2 + src[pos + 2]) >> 2;
 						pos += (inWidth * 4 * 2);
 						bottom = (src[pos] + src[pos + 1] * 2 + src[pos + 2]) >> 2;
-						edge = (abs(left - right) + abs(top - bottom)) >> 1;
-						
+						edge = ((left - right) + (top - bottom)) >> 1;
+
```
Comment (stippi) [inline WonderBrush/src/gui/panels/about/EdgesFilter.cpp:58]: Why is abs() no longer necessary?

**Review by stippi — APPROVED** [review]
Thanks for all your work!

**Pull request merged**